poc/hash_to_field.py

```diff
@@ -92,14 +108,63 @@ def expand_message_xmd(msg, dst, len_in_bytes, hash_fn, security_param):
 
     # compute blocks
     b_vals = [None] * ell
-    b_0 = hash_fn(Z_pad + _as_bytes(msg) + l_i_b_str + I2OSP(0, 1) + dst_prime).digest()
+    msg_prime = Z_pad + _as_bytes(msg) + l_i_b_str + I2OSP(0, 1) + dst_prime
+    b_0 = hash_fn(msg_prime).digest()
     b_vals[0] = hash_fn(b_0 + I2OSP(1, 1) + dst_prime).digest()
     for i in xrange(1, ell):
         b_vals[i] = hash_fn(_strxor(b_0, b_vals[i - 1]) + I2OSP(i + 1, 1) + dst_prime).digest()
 
     # assemble output
-    pseudo_random_octets = (b'').join(b_vals)
-    return pseudo_random_octets[0 : len_in_bytes]
+    pseudo_random_bytes = (b'').join(b_vals)
+    output = pseudo_random_bytes[0 : len_in_bytes]
+
+    vector = {
+        "msg": to_hex(msg),
+        "DST_prime": to_hex(dst_prime),
+        "msg_prime": to_hex(msg_prime),
+        "pseudo_random_bytes": to_hex(output),
+    }
+    result_set.append(vector)
+
+    return output
+
+class Expander(object):
+    test_vectors = []
+
+    def __init__(self, name, dst, hash_fn, security_param):
+        self.name = name
+        self.dst = dst
+        self.hash_fn = hash_fn
+        self.security_param = security_param
+
+    def expand_message(self, msg, len_in_bytes):
+        raise Exception("Not implemented")
+
+    def hash_name(self):
+        return self.hash_fn().name.upper()
+
+    def __dict__(self):
+        return {
+            "name": self.name,
+            "dst": self.dst,
+            "hash": self.hash_name(),
+            "security_param": str(self.security_param),
+            "tests": json.dumps(self.test_vectors),
```

Comment:
I think here is where your are dumping twice.

Reply:
I didn't consider this a big deal, but fixed it nonetheless. Please have a look!